collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/InstanceSummaryDialogFragment.java

```diff
@@ -0,0 +1,144 @@
+package org.odk.collect.android.fragments.dialogs;
+
+import android.content.ContentUris;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.google.android.material.chip.Chip;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.viewmodels.FormMapViewModel;
+import org.odk.collect.android.preferences.AdminKeys;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.provider.InstanceProvider;
+import org.odk.collect.android.provider.InstanceProviderAPI;
+import org.odk.collect.android.utilities.ApplicationConstants;
+import org.odk.collect.android.utilities.IconUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class InstanceSummaryDialogFragment extends BottomSheetDialogFragment {
+    public static final String INSTANCE_ID = "instanceId";
+
+    private long instanceId;
+
+    private FormMapViewModel viewModel;
+
+    @BindView(R.id.submission_name)
+    public TextView submissionName;
+
+    @BindView(R.id.status_icon)
+    public ImageView statusIcon;
+
+    @BindView(R.id.status_text)
+    public TextView statusText;
+
+    @BindView(R.id.info)
+    public TextView infoText;
+
+    @BindView(R.id.openFormChip)
+    public Chip openFormChip;
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        viewModel = ViewModelProviders.of(requireActivity()).get(FormMapViewModel.class);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.submission_summary_layout, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        outState.putLong(INSTANCE_ID, instanceId);
```

Comment:
This is much better and I guess that if truly only necessary data is being saved continuing to use `onSaveInstanceState` is not so bad.

Reply:
This class no longer exists.